core/blockchain/blockchain.go

```diff
@@ -1319,3 +1322,22 @@ func (b *BlockChain) CalcWeight(blocks int64, blockhash *hash.Hash, state byte)
 	}
 	return b.subsidyCache.CalcBlockSubsidy(blocks)
 }
+
+func (b *BlockChain) CheckCacheInvalidTxConfig() error {
+	if b.CacheInvalidTx {
+		hasConfig := true
+		b.db.View(func(dbTx database.Tx) error {
```

Comment:
never call callback like this, compare:
```
b.db.View(func(dbTx database.Tx) error {
   hasconfig = ...
})
if hasconfig ...
```
vs. 
```
err : = b.db.View(func(dbTx database.Tx) error {
   hasconfig = ...
})
if hasconfig ...

```
The 1.) has not guarantee that `hasConfig` has been set **after** you database call and **before** you test it.